### ISSUE TITLE
Failed to obtain commits 404 Not Found issue for Bitbucket collector

### ISSUE THREAD

**Issue by zenglincui** [capitalone/Hygieia#1073]
I got following error after I started Bitbucket collector.  I noticed that both Projects and Pepos are missing in the URL http://mycompany.com/rest/api/1.0/projects//repos//commits?until=Develop&limit=25.  When configuring the Code Repo widget, only Repo URL and Branch are required, I also gave my credentials for the code repo in this case.  

Did I miss something?  Do I need modify the collector code DefaultBitbucketServerClient?
Thanks for any help.
```
C:\Projects\Hygieia-master\collectors\scm\bitbucket\target>java -jar bitbucket-scm-collector-2.0.5-SNAPSHOT.jar --spring.config.location=bitbucket-application.properties
```
```
2017-01-08 12:42:31,707 INFO  com.capitalone.dashboard.Application - Starting Application v2.0.5-SNAPSHOT on PL-JMJWN72 with PID 9196 (C:\Projects\Hygieia-master\col
lectors\scm\bitbucket\target\bitbucket-scm-collector-2.0.5-SNAPSHOT.jar started by scui in C:\Projects\Hygieia-master\collectors\scm\bitbucket\target)
2017-01-08 12:42:31,735 INFO  o.s.c.a.AnnotationConfigApplicationContext - Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@2b9bd
613: startup date [Sun Jan 08 12:42:31 CST 2017]; root of context hierarchy
2017-01-08 12:42:32,422 INFO  o.s.b.f.a.AutowiredAnnotationBeanPostProcessor - JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
2017-01-08 12:42:32,463 INFO  o.h.validator.internal.util.Version - HV000001: Hibernate Validator 5.1.3.Final
2017-01-08 12:42:33,098 INFO  o.s.s.c.ThreadPoolTaskScheduler - Initializing ExecutorService  'taskScheduler'
2017-01-08 12:42:33,164 INFO  c.c.dashboard.config.MongoConfig - ReplicaSetfalse
2017-01-08 12:42:33,165 INFO  c.c.dashboard.config.MongoConfig - Initializing Mongo Client server at: localhost:27017
2017-01-08 12:42:33,521 INFO  org.mongodb.driver.cluster - Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSe
lectionTimeout='30000 ms', maxWaitQueueSize=500}
2017-01-08 12:42:33,568 INFO  c.c.dashboard.config.MongoConfig - Connecting to Mongo: Mongo{options=MongoClientOptions{description='null', readPreference=primary, wr
iteConcern=WriteConcern{w=1, wtimeout=0, fsync=false, j=false, codecRegistry=org.bson.codecs.configuration.ProvidersCodecRegistry@dddebb21, minConnectionsPerHost=0,
maxConnectionsPerHost=100, threadsAllowedToBlockForConnectionMultiplier=5, serverSelectionTimeout=30000, maxWaitTime=120000, maxConnectionIdleTime=0, maxConnectionLi
feTime=0, connectTimeout=10000, socketTimeout=0, socketKeepAlive=false, sslEnabled=false, sslInvalidHostNamesAllowed=false, alwaysUseMBeans=false, heartbeatFrequency
=10000, minHeartbeatFrequency=500, heartbeatConnectTimeout=20000, heartbeatSocketTimeout=20000, localThreshold=15, requiredReplicaSetName='null', dbDecoderFactory=co
m.mongodb.DefaultDBDecoder$1@1232bda6, dbEncoderFactory=com.mongodb.DefaultDBEncoder$1@26bb0cd0, socketFactory=javax.net.DefaultSocketFactory@4108e5e4, cursorFinaliz
erEnabled=true, connectionPoolSettings=ConnectionPoolSettings{maxSize=100, minSize=0, maxWaitQueueSize=500, maxWaitTimeMS=120000, maxConnectionLifeTimeMS=0, maxConne
ctionIdleTimeMS=0, maintenanceInitialDelayMS=0, maintenanceFrequencyMS=60000}, socketSettings=SocketSettings{connectTimeoutMS=10000, readTimeoutMS=0, keepAlive=false
, receiveBufferSize=0, sendBufferSize=0}, serverSettings=ServerSettings{heartbeatFrequencyMS=10000, minHeartbeatFrequencyMS=500}, heartbeatSocketSettings=SocketSetti
ngs{connectTimeoutMS=20000, readTimeoutMS=20000, keepAlive=false, receiveBufferSize=0, sendBufferSize=0}}}
2017-01-08 12:42:33,791 INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:1, serverValue:75}] to localhost:27017
2017-01-08 12:42:33,792 INFO  org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017
, type=STANDALONE, state=CONNECTED, ok=true, version=ServerVersion{versionList=[3, 2, 7]}, minWireVersion=0, maxWireVersion=4, electionId=null, maxDocumentSize=16777
216, roundTripTimeNanos=567771}
2017-01-08 12:42:33,907 INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:2, serverValue:76}] to localhost:27017
2017-01-08 12:42:34,065 INFO  c.c.dashboard.config.MongoConfig - ReplicaSetfalse
2017-01-08 12:42:34,065 INFO  c.c.dashboard.config.MongoConfig - Initializing Mongo Client server at: localhost:27017
2017-01-08 12:42:34,066 INFO  org.mongodb.driver.cluster - Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSe
lectionTimeout='30000 ms', maxWaitQueueSize=500}
2017-01-08 12:42:34,067 INFO  c.c.dashboard.config.MongoConfig - Connecting to Mongo: Mongo{options=MongoClientOptions{description='null', readPreference=primary, wr
iteConcern=WriteConcern{w=1, wtimeout=0, fsync=false, j=false, codecRegistry=org.bson.codecs.configuration.ProvidersCodecRegistry@dddebb21, minConnectionsPerHost=0,
maxConnectionsPerHost=100, threadsAllowedToBlockForConnectionMultiplier=5, serverSelectionTimeout=30000, maxWaitTime=120000, maxConnectionIdleTime=0, maxConnectionLi
feTime=0, connectTimeout=10000, socketTimeout=0, socketKeepAlive=false, sslEnabled=false, sslInvalidHostNamesAllowed=false, alwaysUseMBeans=false, heartbeatFrequency
=10000, minHeartbeatFrequency=500, heartbeatConnectTimeout=20000, heartbeatSocketTimeout=20000, localThreshold=15, requiredReplicaSetName='null', dbDecoderFactory=co
m.mongodb.DefaultDBDecoder$1@1232bda6, dbEncoderFactory=com.mongodb.DefaultDBEncoder$1@26bb0cd0, socketFactory=javax.net.DefaultSocketFactory@4108e5e4, cursorFinaliz
erEnabled=true, connectionPoolSettings=ConnectionPoolSettings{maxSize=100, minSize=0, maxWaitQueueSize=500, maxWaitTimeMS=120000, maxConnectionLifeTimeMS=0, maxConne
ctionIdleTimeMS=0, maintenanceInitialDelayMS=0, maintenanceFrequencyMS=60000}, socketSettings=SocketSettings{connectTimeoutMS=10000, readTimeoutMS=0, keepAlive=false
, receiveBufferSize=0, sendBufferSize=0}, serverSettings=ServerSettings{heartbeatFrequencyMS=10000, minHeartbeatFrequencyMS=500}, heartbeatSocketSettings=SocketSetti
ngs{connectTimeoutMS=20000, readTimeoutMS=20000, keepAlive=false, receiveBufferSize=0, sendBufferSize=0}}}
2017-01-08 12:42:34,079 INFO  c.c.dashboard.config.MongoConfig - ReplicaSetfalse
2017-01-08 12:42:34,079 INFO  c.c.dashboard.config.MongoConfig - Initializing Mongo Client server at: localhost:27017
2017-01-08 12:42:34,079 INFO  org.mongodb.driver.cluster - Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSe
lectionTimeout='30000 ms', maxWaitQueueSize=500}
2017-01-08 12:42:34,080 INFO  c.c.dashboard.config.MongoConfig - Connecting to Mongo: Mongo{options=MongoClientOptions{description='null', readPreference=primary, wr
iteConcern=WriteConcern{w=1, wtimeout=0, fsync=false, j=false, codecRegistry=org.bson.codecs.configuration.ProvidersCodecRegistry@dddebb21, minConnectionsPerHost=0,
maxConnectionsPerHost=100, threadsAllowedToBlockForConnectionMultiplier=5, serverSelectionTimeout=30000, maxWaitTime=120000, maxConnectionIdleTime=0, maxConnectionLi
feTime=0, connectTimeout=10000, socketTimeout=0, socketKeepAlive=false, sslEnabled=false, sslInvalidHostNamesAllowed=false, alwaysUseMBeans=false, heartbeatFrequency
=10000, minHeartbeatFrequency=500, heartbeatConnectTimeout=20000, heartbeatSocketTimeout=20000, localThreshold=15, requiredReplicaSetName='null', dbDecoderFactory=co
m.mongodb.DefaultDBDecoder$1@1232bda6, dbEncoderFactory=com.mongodb.DefaultDBEncoder$1@26bb0cd0, socketFactory=javax.net.DefaultSocketFactory@4108e5e4, cursorFinaliz
erEnabled=true, connectionPoolSettings=ConnectionPoolSettings{maxSize=100, minSize=0, maxWaitQueueSize=500, maxWaitTimeMS=120000, maxConnectionLifeTimeMS=0, maxConne
ctionIdleTimeMS=0, maintenanceInitialDelayMS=0, maintenanceFrequencyMS=60000}, socketSettings=SocketSettings{connectTimeoutMS=10000, readTimeoutMS=0, keepAlive=false
, receiveBufferSize=0, sendBufferSize=0}, serverSettings=ServerSettings{heartbeatFrequencyMS=10000, minHeartbeatFrequencyMS=500}, heartbeatSocketSettings=SocketSetti
ngs{connectTimeoutMS=20000, readTimeoutMS=20000, keepAlive=false, receiveBufferSize=0, sendBufferSize=0}}}
2017-01-08 12:42:34,082 INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:3, serverValue:77}] to localhost:27017
2017-01-08 12:42:34,083 INFO  org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017
, type=STANDALONE, state=CONNECTED, ok=true, version=ServerVersion{versionList=[3, 2, 7]}, minWireVersion=0, maxWireVersion=4, electionId=null, maxDocumentSize=16777
216, roundTripTimeNanos=370939}
2017-01-08 12:42:34,101 INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:4, serverValue:78}] to localhost:27017
2017-01-08 12:42:34,102 INFO  org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017
, type=STANDALONE, state=CONNECTED, ok=true, version=ServerVersion{versionList=[3, 2, 7]}, minWireVersion=0, maxWireVersion=4, electionId=null, maxDocumentSize=16777
216, roundTripTimeNanos=743710}
2017-01-08 12:42:34,295 INFO  c.c.dashboard.config.MongoConfig - ReplicaSetfalse
2017-01-08 12:42:34,295 INFO  c.c.dashboard.config.MongoConfig - Initializing Mongo Client server at: localhost:27017
2017-01-08 12:42:34,296 INFO  org.mongodb.driver.cluster - Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSe
lectionTimeout='30000 ms', maxWaitQueueSize=500}
2017-01-08 12:42:34,306 INFO  c.c.dashboard.config.MongoConfig - Connecting to Mongo: Mongo{options=MongoClientOptions{description='null', readPreference=primary, wr
iteConcern=WriteConcern{w=1, wtimeout=0, fsync=false, j=false, codecRegistry=org.bson.codecs.configuration.ProvidersCodecRegistry@dddebb21, minConnectionsPerHost=0,
maxConnectionsPerHost=100, threadsAllowedToBlockForConnectionMultiplier=5, serverSelectionTimeout=30000, maxWaitTime=120000, maxConnectionIdleTime=0, maxConnectionLi
feTime=0, connectTimeout=10000, socketTimeout=0, socketKeepAlive=false, sslEnabled=false, sslInvalidHostNamesAllowed=false, alwaysUseMBeans=false, heartbeatFrequency
=10000, minHeartbeatFrequency=500, heartbeatConnectTimeout=20000, heartbeatSocketTimeout=20000, localThreshold=15, requiredReplicaSetName='null', dbDecoderFactory=co
m.mongodb.DefaultDBDecoder$1@1232bda6, dbEncoderFactory=com.mongodb.DefaultDBEncoder$1@26bb0cd0, socketFactory=javax.net.DefaultSocketFactory@4108e5e4, cursorFinaliz
erEnabled=true, connectionPoolSettings=ConnectionPoolSettings{maxSize=100, minSize=0, maxWaitQueueSize=500, maxWaitTimeMS=120000, maxConnectionLifeTimeMS=0, maxConne
ctionIdleTimeMS=0, maintenanceInitialDelayMS=0, maintenanceFrequencyMS=60000}, socketSettings=SocketSettings{connectTimeoutMS=10000, readTimeoutMS=0, keepAlive=false
, receiveBufferSize=0, sendBufferSize=0}, serverSettings=ServerSettings{heartbeatFrequencyMS=10000, minHeartbeatFrequencyMS=500}, heartbeatSocketSettings=SocketSetti
ngs{connectTimeoutMS=20000, readTimeoutMS=20000, keepAlive=false, receiveBufferSize=0, sendBufferSize=0}}}
2017-01-08 12:42:34,333 INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:5, serverValue:79}] to localhost:27017
2017-01-08 12:42:34,334 INFO  org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017
, type=STANDALONE, state=CONNECTED, ok=true, version=ServerVersion{versionList=[3, 2, 7]}, minWireVersion=0, maxWireVersion=4, electionId=null, maxDocumentSize=16777
216, roundTripTimeNanos=409792}
2017-01-08 12:42:34,600 INFO  o.s.j.e.a.AnnotationMBeanExporter - Registering beans for JMX exposure on startup
2017-01-08 12:42:34,609 INFO  com.capitalone.dashboard.Application - Started Application in 3.057 seconds (JVM running for 3.588)
2017-01-08 12:45:00,016 INFO  c.c.d.collector.CollectorTask - Running Collector: Bitbucket
2017-01-08 12:45:00,064 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:45:00,064 INFO  c.c.d.collector.CollectorTask - Starting...
2017-01-08 12:45:00,064 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:45:00,092 INFO  c.c.d.collector.CollectorTask - Repo Count              0        0s
2017-01-08 12:45:00,092 INFO  c.c.d.collector.CollectorTask - New Commits             0        0s
2017-01-08 12:45:00,092 INFO  c.c.d.collector.CollectorTask - Finished                         0s
2017-01-08 12:50:00,030 INFO  c.c.d.collector.CollectorTask - Running Collector: Bitbucket
2017-01-08 12:50:00,041 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:50:00,042 INFO  c.c.d.collector.CollectorTask - Starting...
2017-01-08 12:50:00,043 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:50:00,266 ERROR c.c.d.c.DefaultBitbucketServerClient - Failed to obtain commits from http://mycompany.com/rest/api/1.0/projects//repos//commits?until=Develop&limit=25
org.springframework.web.client.HttpClientErrorException: 404 Not Found
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:91) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:614) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:570) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:545) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:466) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at com.capitalone.dashboard.collector.DefaultBitbucketServerClient.makeRestCall(DefaultBitbucketServerClient.java:227) ~[bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.DefaultBitbucketServerClient.getCommits(DefaultBitbucketServerClient.java:88) ~[bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.GitCollectorTask.collect(GitCollectorTask.java:132) [bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.CollectorTask.run(CollectorTask.java:63) [core-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:81) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_77]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_77]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_77]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_77]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_77]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_77]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_77]
2017-01-08 12:50:00,270 INFO  c.c.d.collector.CollectorTask - Repo Count              1        0s
2017-01-08 12:50:00,270 INFO  c.c.d.collector.CollectorTask - New Commits             0        0s
2017-01-08 12:50:00,270 INFO  c.c.d.collector.CollectorTask - Finished                         0s
2017-01-08 12:55:00,013 INFO  c.c.d.collector.CollectorTask - Running Collector: Bitbucket
2017-01-08 12:55:00,017 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:55:00,018 INFO  c.c.d.collector.CollectorTask - Starting...
2017-01-08 12:55:00,018 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-08 12:55:00,088 ERROR c.c.d.c.DefaultBitbucketServerClient - Failed to obtain commits from http://mycompany.com/rest/api/1.0/projects//repos//commits?until=Develop&limit=25
org.springframework.web.client.HttpClientErrorException: 404 Not Found
...
```

**Comment by MarkRx** [comment]
Did you manage to save a widget with a bad git url?

**Comment by zenglincui** [comment]
MarkRx,

Thank you very much for the help!

That was the problem!
When I configured the Code Repo widget, Code Repo's host was used.  
After changed to Code Repo's URL and with Credentials, now I got both project and repo displayed.
    
I tried both in the Code Repo Widget
    http://mylogin@mybitbucketrepo.com/scm/ng/ng-scp.git	
    http://mybitbucketrepo.com/scm/ng/ng-scp.git	
However, I still get error as show below the property file.  

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
My bitbucket property file:

dbname=dashboard
dbusername=dashboarduser
dbpassword=dashboarduser
dbhost=localhost
dbport=27017
dbreplicaset=false
dbhostport=localhost:27017
server.contextPath=/api
server.port=8080

# Logging File location
logging.file=./logs/bitbucket.log

# Collector schedule (required)
git.cron=0 0/5 * * * *

# mandatory:
git.host=mybitbucketrepo.com/
git.api=/rest/api/1.0/

# Maximum number of days to go back in time when fetching commits
git.commitThresholdDays=15

# Page size for rest calls.
#   Only applicable to Bitbucket Server.
#   Only applicable to Bitbucket Cloud.
git.pageSize=25

# Bitbucket product
#   Set to "cloud" to use Bitbucket Cloud (formerly known as Bitbucket)
#   Set to "server" to use Bitbucket Server (formerly known as Stash)
#   More information can be found here:
#     https://github.com/capitalone/Hygieia/issues/609
git.product=server
git.key=DgHaE4mSYqQ7T8QBtrqeEM6eKQHyBIVk

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Bitbucket logs:

2017-01-09 10:20:00,185 ERROR c.c.d.c.DefaultBitbucketServerClient - Failed to obtain commits from http://mybitbucketrepo.com/rest/api/1.0/projects/ABC/repos/XYZ/commits?until=Develop&limit=25
org.springframework.web.client.HttpClientErrorException: 404 Not Found
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:91) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:614) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:570) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:545) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:466) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at com.capitalone.dashboard.collector.DefaultBitbucketServerClient.makeRestCall(DefaultBitbucketServerClient.java:227) ~[bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.DefaultBitbucketServerClient.getCommits(DefaultBitbucketServerClient.java:88) ~[bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.GitCollectorTask.collect(GitCollectorTask.java:132) [bitbucket-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at com.capitalone.dashboard.collector.CollectorTask.run(CollectorTask.java:63) [core-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:81) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_77]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_77]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_77]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_77]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_77]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_77]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_77]
2017-01-09 10:20:00,199 INFO  c.c.d.collector.CollectorTask - Repo Count              1        0s
2017-01-09 10:20:00,200 INFO  c.c.d.collector.CollectorTask - New Commits             0        0s
2017-01-09 10:20:00,201 INFO  c.c.d.collector.CollectorTask - Finished                         0s

**Comment by MarkRx** [comment]
What version is your bitbucket server? Is it behind a proxy?

Hit this url in a browser:
http://mybitbucketrepo.com/rest/api/1.0/projects/ABC/repos/XYZ/commits?until=Develop&limit=25

**Comment by zenglincui** [comment]
My Bitbucket server uses Atlassian Bitbucket v4.2.0.  Is it behind a proxy?  I need find it later. 

Also, I found the branch name should be develop (lowercase), instead of Develop.
However, when I directly hit URL http://mybitbucketrepo.com/rest/api/1.0/projects/ABC/repos/XYZ/commits?until=develop&limit=25, I got 
Unable to open http://mybitbucketrepo.com/rest/api/1.0/projects/ABC/repos/XYZ/commits?until=develop&limit=25.  Cannot download the information you requested.

If I copy the URL to browser, I can get all the data.

After I reconfigured Code Repo widget with correct Branch name, now collector has no error anymore, but just stuck there:

2017-01-09 11:30:46,693 INFO  o.s.j.e.a.AnnotationMBeanExporter - Registering beans for JMX exposure on startup
2017-01-09 11:30:46,701 INFO  com.capitalone.dashboard.Application - Started Application in 4.05 seconds (JVM running for 4.715)
2017-01-09 11:35:00,013 INFO  c.c.d.collector.CollectorTask - Running Collector: Bitbucket
2017-01-09 11:35:00,015 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2017-01-09 11:35:00,016 INFO  c.c.d.collector.CollectorTask - Starting...
2017-01-09 11:35:00,016 INFO  c.c.d.collector.CollectorTask - -----------------------------------
